app/controllers/teachers/progress_reports_controller.rb

```diff
@@ -33,6 +33,194 @@ def demo
     end
   end
 
+  def admin_demo
+    # Create demo school
+    name = params[:name] || 'Admin Demo School'
+    email_safe_school_name = name.gsub(/^a-zA-Z\d/, '').gsub(/\s/, '').downcase
+    teacher_email = "hello+demoadmin-#{email_safe_school_name}@quill.org"
+
+    # If this demo account already exists, just go there
+    existing_admin_user = User.find_by(email: teacher_email)
+    if existing_admin_user.present?
+      sign_out if current_user
+      sign_in existing_admin_user
+      return redirect_to teachers_admin_dashboard_path
+    end
+
```

Comment:
Lets move all of the account creation logic to a module as we have with the Teacher demo.